drivers/mpsl/temp_nrf5/temp_nrf5_mpsl.c

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_nrf_temp
+
+#include <device.h>
+#include <drivers/sensor.h>
+#include <logging/log.h>
+#include <mpsl_temp.h>
+
+LOG_MODULE_REGISTER(temp_nrf5_mpsl, CONFIG_SENSOR_LOG_LEVEL);
+
+/* The MPSL temperature API returns measurements in 0.25C
+ * increments. Scale to mDegrees C.
+ */
+#define TEMP_NRF5_MPSL_TEMP_SCALE (1000000 / 4)
+
+struct temp_nrf5_mpsl_data {
+	int32_t sample;
+};
+
+static int temp_nrf5_mpsl_sample_fetch(struct device *dev,
+				       enum sensor_channel chan)
+{
+	struct temp_nrf5_mpsl_data *data = dev->data;
+
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_DIE_TEMP) {
+		return -ENOTSUP;
+	}
+
+	data->sample = mpsl_temperature_get();
+	LOG_DBG("sample: %d", data->sample);
+
+	return 0;
+}
+
+static int temp_nrf5_mpsl_channel_get(struct device *dev,
+				      enum sensor_channel chan,
+				      struct sensor_value *val)
+{
+	struct temp_nrf5_mpsl_data *data = dev->data;
+	int32_t uval;
+
+	if (chan != SENSOR_CHAN_DIE_TEMP) {
+		return -ENOTSUP;
+	}
+
+	uval = data->sample * TEMP_NRF5_MPSL_TEMP_SCALE;
```

Comment:
You do not really need to multiply by `TEMP_NRF5_MPSL_TEMP_SCALE` and then later device by the same factors. We should try to avoid using the expensive operations `/` and `%`